### ISSUE TITLE
Details List: List items don't announce position in set

### ISSUE THREAD

**Issue by bcoard** [microsoft/fluentui#11819]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)

### Describe the issue:

When a Narrator user navigates to a list item in a Details List, the list item doesn't announce position in set.

### Please provide a reproduction of the issue in a codepen:

1. Navigate to [this](https://developer.microsoft.com/en-us/fabric#/controls/web/detailslist/grouped) page.
2. Turn Narrator on.
3. Using keyboard navigation go to a list item.

**Comment by khmakoto** [comment]
Aggregated into #15355.
